flink-table-store-hive/src/main/java/org/apache/flink/table/store/hive/HiveSchema.java

```diff
@@ -73,11 +90,134 @@ public static HiveSchema extract(Properties properties) {
         String columnTypes = properties.getProperty(serdeConstants.LIST_COLUMN_TYPES);
         List<TypeInfo> typeInfos = TypeInfoUtils.getTypeInfosFromTypeString(columnTypes);
 
+        List<String> partitionKeys = new ArrayList<>();
+        if (properties.containsKey(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS)) {
```

Comment:
I am thinking why we need to check it.
Can we just assert that there is no partition and column definition?

Reply:
Why not? We can support partitions now that we have the real schema.

Reply:
I think the better way is: define any columns is not allowed in hive external table.
Otherwise, the two sides are not aligned and there is a lot of behavior that needs to be explained.